toucan_connectors/rok/rok_connector.py

```diff
@@ -48,6 +51,20 @@ class RokConnector(ToucanConnector):
     def _retrieve_data(self, data_source: RokDataSource) -> pd.DataFrame:
         # Endpoint depends on the authentication mode
         endpoint = f'{self.host}/graphql'
+        date_viewid_parameters = {
+            'start_date': data_source.start_date,
+            'end_date': data_source.end_date,
+            'viewId': data_source.viewId,
+        }
+
+        if data_source.parameters:
+            data_source.query = nosql_apply_parameters_to_query(
+                data_source.query, {**data_source.parameters, **date_viewid_parameters}
+            )
+        else:
+            data_source.query = nosql_apply_parameters_to_query(
+                data_source.query, date_viewid_parameters
+            )
```

Comment:
Nice!
If you want a shortcut for that, you may use:
```suggestion
        if data_source.parameters:
            parameters = {**data_source.parameters, **date_viewid_parameters}
        else:
            parameters = date_viewid_parameters
         data_source.query = nosql_apply_parameters_to_query(
                data_source.query, parameters
            )
```
to centralize the call to the interpolation function

Reply:
Why make it complicated when it can be as simple as that :). Thanks !